packages/babel-parser/src/tokenizer/index.js

```diff
@@ -1024,8 +1025,11 @@ export default class Tokenizer extends LocationParser {
 
     if (this.hasPlugin("bigInt")) {
       if (next === charCodes.lowercaseN) {
-        // disallow floats and legacy octal syntax, new style octal ("0o") is handled in this.readRadixNumber
-        if (isFloat || octal) this.raise(start, "Invalid BigIntLiteral");
+        // disallow floats, legacy octal syntax and non octal decimals
+        // new style octal ("0o") is handled in this.readRadixNumber
+        if (isFloat || octal || isNonOctalDecimal) {
```

Comment:
Does `isNonOctalDecimal == true` _imply_ `(octal || isNonOctalDecimal) == true` ? If so we could consider removing `octal` check here.